src/test/kotlin/com/example/iotmanagementdevice/repository/UserQueryRepositoryTest.kt

```diff
@@ -0,0 +1,244 @@
+        // Given
+        val user = UserFixture.createUser()
+        userQueryRepository.save(user)
+
+        // When
+        val devices = userQueryRepository.findDevicesByUserId(user.id!!)
+
+        // Then
+        assertTrue(devices.isEmpty(), "Expected no devices for the user")
+    }
+
+    @Test
+    fun `should return false when assigning a device to a non-existent user`() {
+        // Given
+        val userId = ObjectId() // Non-existent user ID
+        val deviceId = UserFixture.createDevice().id!!
+
+        // When
+        val result = userQueryRepository.assignDeviceToUser(userId, deviceId)
+
+        // Then
+        assertFalse(result, "Expected false when assigning device to non-existent user")
+    }
+
+    @Test
+    fun `should find all users when multiple users are saved`() {
+        // Given
+        val user1 = UserFixture.createUser().copy(name = "User1", email = "user1@example.com")
+        val user2 = UserFixture.createUser().copy(name = "User2", email = "user2@example.com")
+        userQueryRepository.save(user1)
+        userQueryRepository.save(user2)
+
+        // When
+        val users = userQueryRepository.findAll()
+
+        // Then
+        val expectedUsers = listOf(user1, user2)
+        assertTrue(users.containsAll(expectedUsers), "Expected users not found in the repository")
+    }
+
+    @Test
+    fun `should reflect the device assignment on the user`() {
+        // Given
+        val user = UserFixture.createUser()
+        val device = UserFixture.createDevice()
+
+        userQueryRepository.save(user)
+        deviceQueryRepository.save(device)
+
+        // When
+        userQueryRepository.assignDeviceToUser(user.id!!, device.id!!)
+
+        // Then
+        val updatedDevice = deviceQueryRepository.findById(device.id!!)
+        assertEquals(user.id, updatedDevice?.userId)
+    }
+
+    @Test
+    fun `should reflect the user update after being updated`() {
+        // Given
+        val user = UserFixture.createUser()
+        userQueryRepository.save(user)
+
+        val updatedUserDto = UserUpdateRequestDto(
+            name = "Updated Name",
+            email = "updated@example.com",
+            phoneNumber = "123-456-7890",
+            userPassword = "newPassword"
+        )
+
+        val updatedUser = user.copy(
+            name = updatedUserDto.name,
+            email = updatedUserDto.email,
+            phoneNumber = updatedUserDto.phoneNumber,
+            userPassword = updatedUserDto.userPassword
+        )
+
+        // When
+        userQueryRepository.save(updatedUser)
+
+        // Then
+        val foundUser = userQueryRepository.findById(user.id!!)
+        assertEquals(updatedUserDto.name, foundUser?.name)
+        assertEquals(updatedUserDto.email, foundUser?.email)
+        assertEquals(updatedUserDto.phoneNumber, foundUser?.phoneNumber)
+        assertEquals(updatedUserDto.userPassword, foundUser?.userPassword)
```

Comment:
Why not comparing objects?

Reply:
UpdateDto have different fields than model